src/server/game/Spells/Spell.cpp

```diff
@@ -5831,46 +5831,69 @@ SpellCastResult Spell::CheckRange(bool strict)
     if (!strict && m_casttime == 0)
```

Comment:
I edited the code example included in the previous comment. I was trying to say, that if the target is the caster, then (my understanding is) it will always be within range. Thus

```c++
if (target == m_caster)
    return SPELL_CAST_OK;
```

Removing the need for the check on line 5844